src/main/webapp/app/programming/shared/utils/programming-exercise.utils.spec.ts

```diff
@@ -96,68 +96,71 @@ describe('ProgrammingExerciseUtils', () => {
     describe('isResultPreliminary', () => {
         let result: Result;
         let exercise: ProgrammingExercise;
+        let participation: ProgrammingExerciseStudentParticipation;
 
         beforeEach(() => {
             result = new Result();
             exercise = new ProgrammingExercise(undefined, undefined);
+            participation = new ProgrammingExerciseStudentParticipation();
             exercise.assessmentType = AssessmentType.AUTOMATIC;
         });
 
         it('returns false on undefined exercise', () => {
-            expect(isResultPreliminary(result, undefined)).toBeFalse();
+            expect(isResultPreliminary(result, participation, undefined)).toBeFalse();
```

Comment:
I think we should overload `isResultPreliminary` instead, why do we add complexity here? 

It seems like there are quite some cases where we do not need the participation to retrieve this information, so we should not be forced to pass (and before retrieve a participation) to get the information